### ISSUE TITLE
vue/no-v-for-template-key this rule seems to have outdated errors output for Vue3

### ISSUE THREAD

**Issue by S-Traut** [vuejs/eslint-plugin-vue#1757]
**What rule do you want to change?**
vue/no-v-for-template-key

**Does this change cause the rule to produce more or fewer warnings?**
In my vue3 setup, when I'm trying to add a key to a template, eslint doesn't accept it, therefore, when I try to play the key in the child tags of that template, vueJS doesn't accept the changes and tells me to use the key on the template instead

**How will the change be implemented? (New option, new default behavior, etc.)?**
Maybe this rule is outdated with vue 3

**Please provide some example code that this change will affect:**

Vue accept the following code but eslint throw an error
```vue
<template v-for="(step, key) in getSteps()" :key="key">
  <a href="#"  :class="{ current: isCurrent(step), valid: isValid(step) }">{{ step.short }}</a>
  <span v-if="key < getSteps().length - 1" class="hr"></span>
 </template>
```

Eslint accept the following code, but vueJS crash and tells me to use the key on template instead
```vue
<template v-for="(step, key) in getSteps()">
  <a :key="key" href="#" :class="{ current: isCurrent(step), valid: isValid(step) }">{{ step.short }}</a>
  <span  :key="key" v-if="key < getSteps().length - 1" class="hr" ></span>
 </template>
```

**What will the rule do after it's changed?**
The rule should accept the key on templates

**Comment by ota-meshi** [comment]
`vue/no-v-for-template-key` is a rule for Vue2. So, it is not included in the preset for Vue3.
Perhaps you are using a preset for Vue2 or have you turned ON the rule yourself. You need to change the configuration to turn OFF the `vue/no-v-for-template-key` rule.
